### PR TITLE
Address issue #5126 and a bug in the ExternalInterface

### PR DESCRIPTION
**Pull request by tsnouidui**
## Pull request overview

This pull request repairs the github issue #5126. It also addresses a bug which was causing the external interface to not work properly when using erl variables.

### CODE REVIEW

**Comment by tsnouidui** [comment]
@Myoldmopar: When will this be merged?

File: src/EnergyPlus/EMSManager.cc [hunk 1]
```diff
@@ -692,6 +702,22 @@ namespace EMSManager {
 		GetObjectDefMaxArgs( cCurrentModuleObject, TotalArgs, NumAlphas, NumNums );
 		MaxNumNumbers = max( MaxNumNumbers, NumNums );
 		MaxNumAlphas = max( MaxNumAlphas, NumAlphas );
+		cCurrentModuleObject = "ExternalInterface:FunctionalMockupUnitImport:To:Variable";
```
Comment (Myoldmopar) [inline src/EnergyPlus/EMSManager.cc:705]: This is the only functional code change for the ERL variable issue, right?

File: src/EnergyPlus/RuntimeLanguageProcessor.cc [hunk 2]
```diff
@@ -2705,6 +2705,22 @@ namespace RuntimeLanguageProcessor {
 			GetObjectDefMaxArgs( cCurrentModuleObject, TotalArgs, NumAlphas, NumNums );
 			MaxNumNumbers = max( MaxNumNumbers, NumNums );
 			MaxNumAlphas = max( MaxNumAlphas, NumAlphas );
+			cCurrentModuleObject = "ExternalInterface:FunctionalMockupUnitImport:To:Variable";
```
Comment (Myoldmopar) [inline src/EnergyPlus/RuntimeLanguageProcessor.cc:2708]: Or I guess this goes along with the above fix for ERL variable issues?

File: src/EnergyPlus/ScheduleManager.cc [hunk 3]
```diff
@@ -440,23 +440,23 @@ namespace ScheduleManager {
 		CurrentModuleObject = "ExternalInterface:Schedule";
 		NumExternalInterfaceSchedules = GetNumObjectsFound( CurrentModuleObject );
 		// added for FMI
-		if ( NumCptSchedules > 0 ) {
+		if (NumExternalInterfaceSchedules > 0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/ScheduleManager.cc]: And then these changes are the fix for #5126 right?

**Comment by Myoldmopar** [comment]
I made a couple minor comments.  You wrapped a bunch of lines of code.  Which is _OK_, but it added quite a few unnecessary changes to the review, and it's hard for me to read through the changes in line wrapping and actual, functional code changes.  

The changes in ScheduleManager also need a whitespace padding fixup in the if block to fit in with the existing style.  I'll make that change, because you also changed the file mode of the files:

image

So I'll fix that at the same time.

And finally, the most important thing, this has an example file [crashing](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-2b802ae8afd1265578234daedac7d45942809ec6-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html).  So, yeah, that'll have to be addressed.

**Comment by Myoldmopar** [comment]
Done.  Checkout the failing test file and then I'll do some more testing.

**Comment by tsnouidui** [comment]
@Myoldmopar Yes to all your questions and thanks.

**Comment by tsnouidui** [comment]
@Myoldmopar Do I need to do anything with the failing test? Do you know why it led to a crash?

**Comment by tsnouidui** [comment]
@Myoldmopar I checked out the test and I think, I see where the problem is coming from. I will need to work on that. So please hold on prior to merging.

Thanks!

Thierry

**Comment by Myoldmopar** [comment]
If you take a look at the Files Changed [tab](https://github.com/NREL/EnergyPlus/pull/5593/files), you can see that the RuntimeLanguageProcessor file had a permissions change again.  I had this type of problem long ago when I was editing files on a partition that was used by both my Linux metal and my Windows VM.  It was an NTFS partition, and I didn't have my fstab setup properly, so everything came across as executable for safety's sake.  Is it possible you are doing something similar?

**Comment by tsnouidui** [comment]
@Myoldmopar I have the same configuration and am indeed doing something similar. Do you remember how you fixed it? By the way. I see 6 tests which were not passed (?) and big diffs which I believe cannot be the consequence of my changes. Should I worry about them?

Thanks!

Thierry

**Comment by tsnouidui** [comment]
@Myoldmopar What is the status on this? Please let me know so I can forget this one ;-). Thanks!

**Comment by Myoldmopar** [comment]
OK, I switched off the executable bit from RuntimeLanguageProcessor.  I currently don't have a clear reason why the bit is being set, but I'll try to remember for you.  I'll merge this in now.

**Comment by tsnouidui** [comment]
@Myoldmopar Thanks.

**Pull request merged**